### PR TITLE
Add/Enhance function comments

### PR DESCRIPTION
**Pull request by manju956**
Signed-off-by: manju956 <manjunath.a.c@intel.com>

### CODE REVIEW

File: tc/sgx/trusted_worker_manager/enclave/kme/ext_work_order_info_kme.h [hunk 1]
```diff
@@ -180,9 +184,21 @@ class ExtWorkOrderInfoKME : public ExtWorkOrderInfoImpl {
...
+    /*
+     * Calculates hash on the encrypted keys, input-data-keys and output-data-keys
+     * @param wo_key_info - Instance of WorkOrderKeyInfo
+     * @param wo_key_info_hash [OUT] - Computed hash value
+    */
     void CalculateWorkOrderKeyInfoHash(WorkOrderKeyInfo wo_key_info,
         ByteArray& wo_key_info_hash);
 
+    /*
+     * Creates JSON document of encryption keys, signing keys
```
Comment (rranjan3) [inline tc/sgx/trusted_worker_manager/enclave/kme/ext_work_order_info_kme.h:197]: Did you intend to say "serialized JSON object" by "JSON document"?

File: tc/sgx/trusted_worker_manager/enclave/kme/ext_work_order_info_kme.h [hunk 2]
```diff
@@ -180,9 +184,21 @@ class ExtWorkOrderInfoKME : public ExtWorkOrderInfoImpl {
...
+     * and data encryption keys
```
Comment (rranjan3) [inline tc/sgx/trusted_worker_manager/enclave/kme/ext_work_order_info_kme.h:198]: Also, why do we put the here and not where the function is defined or maybe declared (in the header)?

File: tc/sgx/trusted_worker_manager/enclave/kme/ext_work_order_info_kme.h (hunk 1 above)
Comment (manju956) [reply tc/sgx/trusted_worker_manager/enclave/kme/ext_work_order_info_kme.h:197]: Yes, will rephrase accordingly

File: tc/sgx/trusted_worker_manager/enclave/kme/ext_work_order_info_kme.h (hunk 2 above)
Comment (manju956) [reply tc/sgx/trusted_worker_manager/enclave/kme/ext_work_order_info_kme.h:198]: The reason to put the function comments in header file is to have function comments consolidated in one place. Now, if all the function comments in this header are moved to cpp file, the size of the source file would grow very large. Hence keeping all the comments in the header itself

Comment (danintel) [reply tc/sgx/trusted_worker_manager/enclave/kme/ext_work_order_info_kme.h:198]: One can argue for having comments in the header file or source file.

But reducing the . cpp source file size is not a consideration. We have gigabytes of storage and compilers are very fast processing source.  Also the comments processed several more times in header files than .cpp source files.

Comment (manojgop) [reply tc/sgx/trusted_worker_manager/enclave/kme/ext_work_order_info_kme.h:198]: We can keep  function comments at single place either header file or .cpp file. In this PR my understanding is the comments were move from .cpp file header file

Comment (manju956) [reply tc/sgx/trusted_worker_manager/enclave/kme/ext_work_order_info_kme.h:198]: > One can argue for having comments in the header file or source file.
> 
> But reducing the . cpp source file size is not a consideration. We have gigabytes of storage and compilers are very fast processing source. Also the comments processed several more times in header files than .cpp source files.

Agree @danintel - The point mentioned regarding size of source file is with respect to code readability.
If all comments are put inside cpp file(specifically in this file function comments are more descriptive), the readability of the file becomes bit difficult, hence moved to header file.

**Pull request merged**